### PR TITLE
[Scaffolder] Add translation to editor toolbar

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Translate the Scaffolder template editor toolbar:
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx [hunk 1]
```diff
@@ -120,32 +131,37 @@ export function TemplateEditorToolbar(props: {
...
+                </li>
+                <li>
+                  {t('templateEditorToolbar.addToCatalogDialogContent.step4')}
+                  <br />
                   <Link to="https://backstage.io/docs/features/software-templates/adding-templates/">
-                    documentation
-                  </Link>{' '}
-                  to register the new template repository in software catalog
+                    https://backstage.io/docs/features/software-templates/adding-templates/
+                  </Link>
```
Comment (Rugvip) [inline plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx:157]: We'll need to make this link customizable for this to be useful imo. I don't think there's any way to do that right now right?

File: plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx [hunk 2]
```diff
@@ -120,32 +131,37 @@ export function TemplateEditorToolbar(props: {
...
             <DialogContentText id="publish-dialog-slide-description">
-              Follow the instructions below to create or update a template:
+              {t(
+                'templateEditorToolbar.addToCatalogDialogContent.introduction',
+              )}
               <ol>
-                <li>Save the template files in a local directory</li>
                 <li>
-                  Create a pull request to a new or existing git repository
+                  {t('templateEditorToolbar.addToCatalogDialogContent.step1')}
```
Comment (Rugvip) [inline plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx:144]: Is it possible to have the separate step be newlines or smth within the translation messages instead? Feeling this a bit too wonky to ship

File: plugins/scaffolder/src/translation.ts [hunk 3]
```diff
@@ -298,5 +298,24 @@ export const scaffolderTranslationRef = createTranslationRef({
...
+        step1: 'Save the template files in a local directory',
+        step2: 'Create a pull request to a new or existing git repository',
+        step3:
+          'If the template already exists, the changes will be reflected in the software catalog once the pull request gets merged',
+        step4:
+          'But if you are creating a new template, follow the documentation linked below to register the new template repository in software catalog',
+      },
+      addToCatalogDialogActions: {
+        closeButton: 'Close',
+      },
```
Comment (Rugvip) [inline plugins/scaffolder/src/translation.ts:318]: Thinking this doesn't need the extra wrapping tbh, something like `addToCatalogDialogCloseButton`?

**Review by Rugvip — CHANGES_REQUESTED** [review]
The dialog content is a bit tricky, thinking it's best to not ship translations for now if there's no clean way to implement it. We can always ship the rest though.

File: plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx (hunk 1 above)
Comment (camilaibs) [reply plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx]: I added the link to the translation keys, is that a safe solution for now?

File: plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx (hunk 2 above)
Comment (camilaibs) [reply plugins/scaffolder/src/alpha/components/TemplateEditorPage/TemplateEditorToolbar.tsx]: Refactored to add a step for each new line.

File: plugins/scaffolder/src/translation.ts (hunk 3 above)
Comment (camilaibs) [reply plugins/scaffolder/src/translation.ts]: It contains more inner keys now, please take a look.

**Review by Rugvip — APPROVED** [review]
Oh that seems like a good approach for what we got right now! 👍 

Let's :shipit: 👌

**Pull request merged**